### PR TITLE
Makefile improvements

### PR DESCRIPTION
**Pull request by hasufell**
see the commit descriptions

### CODE REVIEW

**Comment by hasufell** [comment]
fixes #243 and #264

**Comment by hasufell** [comment]
the patches have already been applied downstream https://github.com/gentoo/gentoo/commit/2dcfb00c893c3190165f76a9b592fcf25b813809
let me know what you think

**Comment by hasufell** [comment]
responsive upstream... not.

**Comment by indutny** [comment]
Yay, please reopen! :)

File: Makefile [hunk 1]
```diff
@@ -101,8 +101,10 @@ test-valgrind: test_g
 libhttp_parser.o: http_parser.c http_parser.h Makefile
 	$(CC) $(CPPFLAGS_FAST) $(CFLAGS_LIB) -c http_parser.c -o libhttp_parser.o
 
-library: libhttp_parser.o
-	$(CC) $(LDFLAGS_LIB) -o $(SONAME) $<
+$(SONAME): libhttp_parser.o
+	$(CC) $(CFLAGS_LIB) $(LDFLAGS_LIB) -o $(SONAME) $<
```
Comment (indutny) [inline Makefile]: Any reason to not use `$@` here?

File: Makefile [hunk 2]
```diff
@@ -55,7 +55,7 @@ CFLAGS_LIB = $(CFLAGS_FAST) -fPIC
 LDFLAGS_LIB = $(LDFLAGS) -shared
 
 INSTALL ?= install
-PREFIX ?= $(DESTDIR)/usr/local
```
Comment (indutny) [inline Makefile]: Where is `DESTDIR` defined?

Comment (indutny) [inline Makefile]: Argh, nevermind.

File: Makefile [hunk 3]
```diff
@@ -123,19 +123,19 @@ tags: http_parser.c http_parser.h test.c
 	ctags $^
 
 install: library
-	$(INSTALL) -D  http_parser.h $(INCLUDEDIR)/http_parser.h
-	$(INSTALL) -D $(SONAME) $(LIBDIR)/$(SONAME)
-	ln -s $(LIBDIR)/$(SONAME) $(LIBDIR)/libhttp_parser.$(SOEXT)
+	$(INSTALL) -D  http_parser.h $(DESTDIR)$(INCLUDEDIR)/http_parser.h
```
Comment (indutny) [inline Makefile]: Shouldn't there be `/` between `DESTDIR` and `INCLUDEDIR`?

**Comment by tgurr** [comment]
We're shipping these patches downstream in Exherbo now as well, they still apply for 2.7.1 and it would be nice to get these merged upstream finally.

**Pull request closed without merging**